lib/graphql/types/int.rb

```diff
@@ -10,7 +10,7 @@ class Int < GraphQL::Schema::Scalar
       MAX = (2**31) - 1
 
       def self.coerce_input(value, _ctx)
-        value.is_a?(Integer) ? value : nil
+        value.is_a?(Integer) && value >= MIN && value <= MAX ? value : nil
```

Comment:
This implementation looks fine spec-wise, but I want to make sure that we provide a path for people who want to preserve the old behavior. How about: 

- create a new error class, `GraphQL::IntegerDecodingError`, which can basically be identical to `GraphQL::IntegerEncodingError`. (But, we'll use a different class so that people can handle the two scenarios differently 
- When the value is out of range, make a new `GraphQL::IntegerDecodingError` and pass it to `ctx.schema.type_error(err, ctx)` 
- Extend the default type error handler (https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/default_type_error.rb) to handle instances of that error class by returning `nil`. (Or, should it raise the error to the client?)

That way: 

- The behavior you've proposed here is the default 
- Users of the library can retain the old behavior by adding to their schema: 

  ```ruby 
  def self.type_error(err, ctx)
    if err.is_a?(GraphQL::IntegerDecodingError)
      err.value # return it anyways, since this is how graphql-ruby used to work 
    else 
      super 
    end 
  end 
  ```


How does that sound to you?

Reply:
> to handle instances of that error class by returning nil. (Or, should it raise the error to the client?)

I thought it should return nil since, since it would be consistent with other scalar types that seem to return nil for invalid inputs in #corece_input.

Reply:
> consistent

makes sense to me :+1: